jwst/resample/resample_spec_step.py

```diff
@@ -45,15 +51,14 @@ def process(self, input):
             kwargs = self._set_spec_defaults()
 
         self.drizpars = kwargs
-
         if isinstance(input_models[0], MultiSlitModel):
```

Comment:
The regtest problem with NIRSpec MOS data in the level-3 pipeline occurs here and is due to the fact that the input models undergo a bunch of remanipulation before getting to this line. First, at the start of `calwebb_spec3` the `MultiSlitModel`s get reorganized into `MultiExposureModel`s by the `exp_to_source` process. Then at some point the container of `MultiExposureModel`s get reorganized into a `SourceModelContainer` and that process converts each model within the container back into a simple `SlitModel`. So the first model in the container, which is what's being tested here, is no longer `MultiSlitModel`, but is instead `SlitModel`, so it doesn't use `process_multislit`. Obviously this worked OK before, but now something has changed in the preceding code such that it doesn't work right anymore.

Reply:
@jdavies-st  any suggestions?

Reply:
Ah, I get it now. Even with the original code NIRSpec MOS data were being sent to the `process_slit` function (as opposed to `process_multislit`), because the data have been reorganized upstream in `calwebb_spec3` and hence each time you call `resample_spec` you're giving it data from only 1 slit. There are still multiple instances of data (from multiple exposures) for that slit, but it's still just 1 slit. So `process_slit` is being fed a container of multiple `SlitModel`s. The `process_multislit` function would only get used when running NIRSpec MOS data through the `calwebb_spec2` pipeline, where you've got a single instance of data for multiple slits.

So having the NIRSpec MOS data go through `process_slit` during `calwebb_spec3` is correct and hence the only way to fix the current situation is to add a check down in `process_slit` so that it only includes the xmin, xmax, ymin, ymax arguments in the call to `do_drizzle` if `input_models[0].meta.exposure.type == "MIR_LRS-FIXEDSLIT"`. Making that change results in successful NIRSpec MOS processing through `calwebb_spec3`.

Wow, this is really complicated stuff ...